src/wp-includes/media.php

```diff
@@ -833,6 +835,14 @@ function image_get_intermediate_size( $post_id, $size = 'thumbnail' ) {
 		$data['url']  = path_join( dirname( $file_url ), $data['file'] );
 	}
 
+	// Use the correct MIME type if available.
+	if ( ! empty( $mime_type ) && isset( $data['sources'][ $mime_type ]['file'] ) && $mime_type !== $data['mime-type'] ) {
+		$data['file']      = str_replace( wp_basename( $data['file'] ), $data['sources'][ $mime_type ]['file'], $data['file'] );
+		$data['path']      = str_replace( wp_basename( $data['path'] ), $data['sources'][ $mime_type ]['file'], $data['path'] );
+		$data['url']       = str_replace( wp_basename( $data['url'] ), $data['sources'][ $mime_type ]['file'], $data['url'] );
+		$data['mime-type'] = $mime_type;
+	}
```

Comment:
Let's in any case unset `$data['sources']` here since it's not part of the expected return type and also doesn't need to be.

Also I don't think there is a need to include `mime-type` in the data as you're doing above, since that's not part of the response either. _If_ we want to add it for some reason, it should be added in all circumstances and also added to the `@return` documentation for this function. But I don't think that's actually needed.